src/EditorFeatures/Core/DocumentationComments/CopilotGenerateDocumentationCommentProvider.cs

```diff
@@ -0,0 +1,306 @@
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.DocumentationComments
+{
+    internal class CopilotGenerateDocumentationCommentProvider : SuggestionProviderBase
+    {
+        private SuggestionManagerBase? _suggestionManager;
+        private readonly ICopilotCodeAnalysisService _copilotService;
+
+        public readonly IThreadingContext ThreadingContext;
+
+        public bool Enabled => _enabled && (_suggestionManager != null);
+        private bool _enabled = true;
+
+        public CopilotGenerateDocumentationCommentProvider(IThreadingContext threadingContext, ICopilotCodeAnalysisService copilotService)
+        {
+            _copilotService = copilotService;
+            ThreadingContext = threadingContext;
+        }
+
+        public async Task InitializeAsync(ITextView textView, SuggestionServiceBase suggestionServiceBase, CancellationToken cancellationToken)
+        {
+            _suggestionManager ??= await suggestionServiceBase.TryRegisterProviderAsync(this, textView, "AmbientAIDocumentationComments", cancellationToken).ConfigureAwait(false);
+        }
+
+        public async Task GenerateDocumentationProposalAsync(DocumentationCommentSnippet snippet,
+        ITextSnapshot oldSnapshot, VirtualSnapshotPoint oldCaret, CancellationToken cancellationToken)
+        {
+            await Task.Yield().ConfigureAwait(false);
+
+            if (!Enabled)
+            {
+                return;
+            }
+
+            var snippetProposal = GetSnippetProposal(snippet.SnippetText, snippet.MemberNode, snippet.Position, snippet.CaretOffset);
+
+            if (snippetProposal is null)
+            {
+                return;
+            }
+
+            // Do not do IntelliCode line completions if we're about to generate a documentation comment
+            // so that won't have interfering grey text.
+            var intellicodeLineCompletionsDisposable = await _suggestionManager!.DisableProviderAsync(SuggestionServiceNames.IntelliCodeLineCompletions, cancellationToken).ConfigureAwait(false);
+
+            var proposalEdits = await GetProposedEditsAsync(snippetProposal, _copilotService, oldSnapshot, snippet.IndentText, cancellationToken).ConfigureAwait(false);
+
+            var proposal = Proposal.TryCreateProposal(null, proposalEdits, oldCaret, flags: ProposalFlags.SingleTabToAccept);
+
+            if (proposal is null)
+            {
+                return;
+            }
+
+            var suggestion = new DocumentationCommentSuggestion(this, proposal, _suggestionManager, intellicodeLineCompletionsDisposable);
+
+            var session = await suggestion.GetSuggestionSessionAsync(cancellationToken).ConfigureAwait(false);
+
+            if (session != null)
+            {
+                await TryDisplaySuggestionAsync(session, suggestion, cancellationToken).ConfigureAwait(false);
+            }
```

Comment:
I think this can be moved into `DocumentationCommentSuggestion` as well. No need to expose session to the provider